src/test/java/org/folio/circulation/resources/RequestCollectionResourceTest.java

```diff
@@ -0,0 +1,118 @@
+package org.folio.circulation.resources;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang.StringUtils;
+import org.folio.circulation.domain.*;
+import org.folio.circulation.infrastructure.storage.ServicePointRepository;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.GetManyRecordsClient;
+import org.folio.circulation.support.http.client.Response;
+import org.folio.circulation.support.results.Result;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class RequestCollectionResourceTest {
+
+  String queryWithOnePSPId = "(effectiveLocationPrimaryServicePointId==\"a9b8247d-3a7a-410e-8df6-1c5af342aa77\") sortby requestDate";
+  String queryWithTwoPSPIds = "(effectiveLocationPrimaryServicePointId==(\"a9b8247d-3a7a-410e-8df6-1c5af342aa77\" or \"3a40852d-49fd-4df2-a1f9-6e2641a6e91f\")) sortby requestDate";
+  String queryWithThreePSPIds = "(effectiveLocationPrimaryServicePointId==(\"a9b8247d-3a7a-410e-8df6-1c5af342aa77\" or \"3a40852d-49fd-4df2-a1f9-6e2641a6e91f\" or \"c4c90014-c8c9-4ade-8f24-b5e313319f4b\")) sortby requestDate";
+  String queryWithNoPSPIds = "() sortby requestDate";
+  String queryWithNoPSPIdsSortByDesc = "() sortby requestDate/sort.descending";
+
+  RequestCollectionResource requestCollectionResource = new RequestCollectionResource(mock(HttpClient.class));
+
+  @Test
+  void parseQueryWithOnePSPId() {
+    List<String> tokens = requestCollectionResource.parseEffectiveLocationIds(queryWithOnePSPId);
+    assertEquals(1, tokens.size());
+    assertEquals("a9b8247d-3a7a-410e-8df6-1c5af342aa77", tokens.get(0));
+  }
+
+  @Test
+  void parseQueryWithTwoPSPIds() {
+    List<String> tokens = requestCollectionResource.parseEffectiveLocationIds(queryWithTwoPSPIds);
+    assertEquals(2, tokens.size());
+    assertEquals("a9b8247d-3a7a-410e-8df6-1c5af342aa77", tokens.get(0));
+    assertEquals("3a40852d-49fd-4df2-a1f9-6e2641a6e91f", tokens.get(1));
+  }
+
+  @Test
+  void parseQueryWithThreePSPIds() {
```

Comment:
Does it make sense to do the same as previous tests do? Maybe it makes sense to consider including inclusion within a single test?